### ISSUE TITLE
Unable to build a project using the org.gwtproject::gwt::2.10.0 bom

### ISSUE THREAD

**Issue by jeffrpowell** [gwtproject/gwt#9753]
**GWT version:** 2.10.0
**Browser (with version):** n/a, maven build issue
**Operating System:** Windows 10

---

##### Description

Unable to build a project using the org.gwtproject::gwt::2.10.0 bom.

This is because the bom's pom references gwt-dev, gwt-servlet, and gwt-user using the com.google.gwt groupId, but 2.10.0 versions of those jars have not been uploaded to maven central under the old groupId.

References:
[https://search.maven.org/search?q=g:com.google.gwt](https://search.maven.org/search?q=g:com.google.gwt)
[https://search.maven.org/remotecontent?filepath=org/gwtproject/gwt/2.10.0/gwt-2.10.0.pom](https://search.maven.org/remotecontent?filepath=org/gwtproject/gwt/2.10.0/gwt-2.10.0.pom)

I expect 2.10.0 jars to be uploaded for the old groupId because of this note written on [https://github.com/gwtproject/gwt/releases/tag/2.10.0](https://github.com/gwtproject/gwt/releases/tag/2.10.0):
`This will be the last published version using the com.google.gwt groupId.`

##### Steps to reproduce

Include the following dependency setup in a maven pom.xml file
```
<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>org.gwtproject</groupId>
			<artifactId>gwt</artifactId>
			<version>2.10.0</version>
			<scope>import</scope>
			<type>pom</type>
		</dependency>
	<dependencies>
</dependencyManagement>
<dependencies>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-servlet</artifactId>
	</dependency>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-user</artifactId>
	</dependency>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-dev</artifactId>
	</dependency>
<dependencies>
```

It will fail with an error message that includes the following:
`Could not resolve dependencies for project [REDACTED]: The following artifacts could not be resolved: com.google.gwt:gwt-servlet:jar:2.10.0, com.google.gwt:gwt-dev:jar:2.10.0, com.google.gwt:gwt-user:jar:2.10.0: Failure to find com.google.gwt:gwt-servlet:jar:2.10.0`

Note the attempt to find **com.google.gwt**:gwt-\*, not the expected org.gwtproject:gwt-*

##### Known workarounds
Individually and explicitly depend on the org.gwtproject 2.10.0 gwt-user, gwt-dev, and gwt-servlet jars without using the bom in a dependencyManagement section.

**Comment by tbroyer** [comment]
The workaround is actually to add **exclusions** for the `com.google.gwt` artifacts… until they're actually deployed.

```xml
<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>org.gwtproject</groupId>
			<artifactId>gwt</artifactId>
			<version>2.10.0</version>
			<scope>import</scope>
			<type>pom</type>
		</dependency>
	</dependencies>
</dependencyManagement>
<dependencies>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-servlet</artifactId>
		<exclusions>
			<exclusion>
				<groupId>com.google.gwt</groupId>
				<artifactId>gwt-servlet</artifactId>
			</exclusion>
		</exclusions>
	</dependency>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-user</artifactId>
		<exclusions>
			<exclusion>
				<groupId>com.google.gwt</groupId>
				<artifactId>gwt-user</artifactId>
			</exclusion>
		</exclusions>
	</dependency>
	<dependency>
		<groupId>org.gwtproject</groupId>
		<artifactId>gwt-dev</artifactId>
		<exclusions>
			<exclusion>
				<groupId>com.google.gwt</groupId>
				<artifactId>gwt-dev</artifactId>
			</exclusion>
		</exclusions>
	</dependency>
</dependencies>
```

**Comment by niloc132** [comment]
The release is up, but it isn't completely ready to go yet, until Google has been able to successfully deploy their portion, as you noted. The first attempt was made, but the version of the Google poms that was written by non-Googlers turned out to not work properly (a review is pending, and we'll be testing it shortly as well).

Our apologies for this extra hassle, but there are still a few steps in the release (including updating org.codehaus.mojo:gwt-maven-plugin) until we're entirely ready to go.

Note that it may be necessary to add exclusions like the above to other dependencies in your project - anything that references com.google.gwt groupIds might require this, until the release is finalized from Google's side. 

This is expected to be the only release that has this hiccup, as we won't be changing groupIds a second time.

**Comment by niloc132** [comment]
@jeffrpowell can you confirm that this is working now that Google has released their 2.10.0 artifacts?

**Comment by jeffrpowell** [comment]
Thanks for the heads-up. Just confirmed that this issue is no longer reproducing for me.
